`LauncherInterceptor` is missing from the `org.junit.platform.launcher` module's `uses` declaration

## Steps to reproduce

- Set junit.platform.launcher.interceptors.enabled to true
- Try running JUnit tests as modules
- Exception: `Caused by: java.util.ServiceConfigurationError: org.junit.platform.launcher.LauncherInterceptor: module org.junit.platform.launcher does not declare uses`

## Context

 - Used versions (Jupiter/Vintage/Platform): Platform 1.10.1
 - Build Tool/IDE: Gradle

Probably an oversight since JUnit platform consumed as a module is an uncommon configuration.
The new Interceptor probably just needs to be added here:
https://github.com/junit-team/junit5/blob/main/junit-platform-launcher/src/module/org.junit.platform.launcher/module-info.java#L37

--- a/documentation/src/docs/asciidoc/release-notes/release-notes-5.11.0-M1.adoc
+++ b/documentation/src/docs/asciidoc/release-notes/release-notes-5.11.0-M1.adoc
@@ -16,6 +16,8 @@ JUnit repository on GitHub.
 ==== Bug Fixes
 
 * ❓
+* Fixes use of the launcher as a Java module when `junit.platform.launcher.interceptors.enabled` is enabled
+  - See link:https://github.com/junit-team/junit5/issues/3561[issue 3561] for details.
 
 ==== Deprecations and Breaking Changes
 

--- a/junit-platform-launcher/src/module/org.junit.platform.launcher/module-info.java
+++ b/junit-platform-launcher/src/module/org.junit.platform.launcher/module-info.java
@@ -16,6 +16,8 @@
  * @since 1.0
  * @uses org.junit.platform.engine.TestEngine
  * @uses org.junit.platform.launcher.LauncherDiscoveryListener
+ * @uses org.junit.platform.launcher.LauncherInterceptor
+ * @uses org.junit.platform.launcher.LauncherSessionListener
  * @uses org.junit.platform.launcher.PostDiscoveryFilter
  * @uses org.junit.platform.launcher.TestExecutionListener
  */
@@ -32,6 +34,7 @@ module org.junit.platform.launcher {
 
 	uses org.junit.platform.engine.TestEngine;
 	uses org.junit.platform.launcher.LauncherDiscoveryListener;
+	uses org.junit.platform.launcher.LauncherInterceptor;
 	uses org.junit.platform.launcher.LauncherSessionListener;
 	uses org.junit.platform.launcher.PostDiscoveryFilter;
 	uses org.junit.platform.launcher.TestExecutionListener;

--- a/platform-tooling-support-tests/projects/jar-describe-module/junit-platform-launcher.expected.txt
+++ b/platform-tooling-support-tests/projects/jar-describe-module/junit-platform-launcher.expected.txt
@@ -10,6 +10,7 @@ requires org.junit.platform.commons transitive
 requires org.junit.platform.engine transitive
 uses org.junit.platform.engine.TestEngine
 uses org.junit.platform.launcher.LauncherDiscoveryListener
+uses org.junit.platform.launcher.LauncherInterceptor
 uses org.junit.platform.launcher.LauncherSessionListener
 uses org.junit.platform.launcher.PostDiscoveryFilter
 uses org.junit.platform.launcher.TestExecutionListener

